drivers/temperature/adt7420/adt7420.c

```diff
@@ -205,15 +272,48 @@ void adt7420_set_operation_mode(struct adt7420_dev *dev,
 void adt7420_set_resolution(struct adt7420_dev *dev,
 			    uint8_t resolution)
 {
-	uint8_t register_value = 0;
+	uint8_t register_value[1] = { 0 };
+	uint8_t device_reg_address;
+
+	if (chip_info[dev->active_device].communication == SPI) {
+		device_reg_address = ADT7320_REG_CONFIG;
+		register_value[0]  = adt7420_get_register_value(dev, device_reg_address);
+	} else {
+		device_reg_address = ADT7420_REG_CONFIG;
+		register_value[0]  = adt7420_get_register_value(dev, device_reg_address);
+	}
+
+	register_value[0] &= ~ADT7420_CONFIG_RESOLUTION;
```

Comment:
All this masking that you do extensively in several functions could be done by calling one function, something like this:
https://github.com/analogdevicesinc/no-OS/blob/master/drivers/adc/ad7606/ad7606.c#L339-L355

Reply:
The masking has been implemented as suggested as a part of the set_register_value() function. Following changes made respective to this:
1.) Added new function parameters for the 'mask' and the 'value' parameter
2.)Removed lines preceding the set_register_value() function call that dealt with masking.